File: c++_programming/all_programs/print_anagrams_together.cpp
#include <iostream>
#include <string>
using namespace std;
class Word
{
	public:
	int index;
	char *str;
};

class Duparray
{
	public:
	int size;
	Word *array;
};

void create_duplicate_array(char *str[],int size)
{
	Duparray *duparray=new Duparray();
	duparray->size=size;
	duparray->array=new Word[(duparray->size*sizeof(Word))];
	int i;
	for(i=0;i<size;i++)
	{
		duparray->array[i].str=new char[(strlen(str[i]+1))];
		duparray->array[i].index=i;
		strcpy(duparray->array[i].str,str[i]);
	}
	
	sort(duparray->array[0],duparray->array[size-1]);
	

	
}


int main()
{
	char *word_arr[]={"cat","dog","tac","god","act"};
	int size=sizeof(word_arr)/sizeof(word_arr[0]);
	create_duplicate_array(word_arr,size);
	return 0;
}

File: python/print_anagrams_together.py
class Word:
    def __init__(self,string,index):
        self.string=string
        self.index=index



def duplicate_array(word,size):
    duplicate=[]
    for i in range(size):
        duplicate.append(Word(word[i],i))
    
    for i in range(size):
        duplicate[i].string=''.join(sorted(duplicate[i].string))
    
    duplicate=sorted(duplicate,key=lambda k:k.string)
    for i in range(size):
        print(word[duplicate[i].index])
    


wordarr=['cat','dog','act','god','tac']
size=len(wordarr)
duplicate_array(wordarr,size)
